examples/json-extension/client/src/extension.ts

```diff
@@ -20,80 +20,92 @@
-      fileEvents: workspace.createFileSystemWatcher("**/.clientrc"),
-    },
-  };
+    return {
+        // Register the server for plain text documents
+        documentSelector: [
+            { scheme: "file", language: "json" },
+            { scheme: "untitled", language: "json" },
+        ],
+        outputChannelName: "[pygls] JsonLanguageServer",
+        synchronize: {
+            // Notify the server about file changes to '.clientrc files contain in the workspace
+            fileEvents: workspace.createFileSystemWatcher("**/.clientrc"),
+        },
+    };
 }
 
 function isStartedInDebugMode(): boolean {
-  return process.env.VSCODE_DEBUG_MODE === "true";
+    return process.env.VSCODE_DEBUG_MODE === "true";
 }
 
-function startLangServerTCP(addr: number): LanguageClient {
-  const serverOptions: ServerOptions = () => {
-    return new Promise((resolve, reject) => {
-      const clientSocket = new net.Socket();
-      clientSocket.connect(addr, "127.0.0.1", () => {
-        resolve({
-          reader: clientSocket,
-          writer: clientSocket,
+function startLangServerTCP(port: number): LanguageClient {
+    const serverOptions: ServerOptions = () => {
+        return new Promise((resolve, reject) => {
+            const clientSocket = new net.Socket();
+            clientSocket.connect(port, "127.0.0.1", () => {
+                resolve({
+                    reader: clientSocket,
+                    writer: clientSocket,
+                });
+            });
         });
-      });
-    });
-  };
+    };
 
-  return new LanguageClient(`tcp lang server (port ${addr})`, serverOptions, getClientOptions());
+    return new LanguageClient(`tcp lang server (port ${port})`, serverOptions, getClientOptions());
 }
 
 function startLangServer(
-  command: string, args: string[], cwd: string,
+    command: string, args: string[], cwd: string,
 ): LanguageClient {
-  const serverOptions: ServerOptions = {
-    args,
-    command,
-    options: { cwd },
-  };
+    const serverOptions: ServerOptions = {
+        args,
+        command,
+        options: { cwd },
+    };
 
-  return new LanguageClient(command, serverOptions, getClientOptions());
+    return new LanguageClient(command, serverOptions, getClientOptions());
+}
+
+function getConfig(key: string): WorkspaceConfiguration {
+    return workspace.getConfiguration(key, editor ? editor.document.uri : null);
+}
+
+function getPython(): string {
+    return getConfig("python").get<string>("pythonPath");
 }
 
 export function activate(context: ExtensionContext) {
-  if (isStartedInDebugMode()) {
-    // Development - Run the server manually
-    client = startLangServerTCP(2087);
-  } else {
-    // Production - Client is going to run the server (for use within `.vsix` package)
-    const cwd = path.join(__dirname, "../");
-    const pythonPath = workspace.getConfiguration("python").get<string>("pythonPath");
+    const debugPort = process.env.DebugTCP;
```

Comment:
Function `isStartedInDebugMode` is not used anymore. I think we should keep port hardcoded and change it in another PR, because we should read that environment variable when starting the server from VS Code.